frostsnap_core/src/coordinator.rs

```diff
@@ -872,6 +875,42 @@ impl FrostCoordinator {
         }])
     }
 
+    pub fn verify_address(
+        &self,
+        access_structure_ref: AccessStructureRef,
+        derivation_index: u32,
+        encryption_key: SymmetricKey,
```

Comment:
We shouldn't need the decryption key to verify an address. You should send the master app key over, the device can verify it's correct with the `KeyId`, and derive the address from the `MasterAppkey`.